### PR TITLE
Tech/alt e2e without verdaccio

### PR DESCRIPTION
**Pull request by ndelangen**
Issue: improving dev experience and CI performance

## What I did

I removed the need for publishing packages
All examples are now setup to have a resolutions section in package.json with file:/ refs as versions.
This forces yarn to install from those locations

### CODE REVIEW

**Comment by gaetanmaisse** [comment]
I think you'll need to delete https://github.com/storybookjs/storybook/blob/next/scripts/run-e2e.ts#L86 and L88 as there is no more local registry

**Comment by ndelangen** [comment]
removed it @gaetanmaisse !

**Comment by merceyz** [comment]
I suggest running all those e2e tests using Yarn 2, using the node-modules linker where needed. You can do all this using **one** command with V2 and you don't have to maintain any list.

**Comment by shilman** [comment]
@merceyz that sounds wonderful. how does it work?

**Comment by ndelangen** [comment]
@merceyz I'd love to work with you to make that happen, as it stands I do not know how to do what you described.

Can the suggestion be applied to this PR?

File: scripts/run-e2e.ts [hunk 1]
```diff
@@ -82,8 +83,6 @@ const configureYarn2 = async ({ cwd }: Options) => {
     // See https://github.com/yarnpkg/berry/pull/2078
     // As soon as a new version of Yarn is released remove next line
     `yarn set version from sources --branch 2078`,
-    // ⚠️ Need to set registry because Yarn 2 is not using the conf of Yarn 1
-    `yarn config set npmScopes --json '{ "storybook": { "npmRegistryServer": "http://localhost:6000/" } }'`,
     // Some required magic to be able to fetch deps from local registry
     `yarn config set unsafeHttpWhitelist --json '["localhost"]'`,
```
Comment (merceyz) [inline scripts/run-e2e.ts:87]: If all the e2e tests ran using Yarn 2 you would add the following command here
```
yarn link "/path/to/storybook" --all
```
Then if the test doesn't support PnP yet run
```
yarn config set nodeLinker node-modules
```

**Comment by merceyz** [comment]
> @merceyz that sounds wonderful. how does it work?

@shilman The link command in V2 can connect the dependencies in a local project to an external one - https://yarnpkg.com/cli/link

**Comment by gaetanmaisse** [comment]
To rework when https://github.com/storybookjs/storybook/pull/13559 will be merged.
Plus, https://github.com/yarnpkg/berry/pull/1971 has been merged on Yarn side, it's great news for the e2e overhaul/simplification.

**Pull request closed without merging**